application-services/custom/camera-management/appcamera/app.go

```diff
@@ -43,11 +44,38 @@ func (app *CameraManagementApp) Run() error {
 		return err
 	}
 
+	// Subscribe to events.
+	err := app.service.SetDefaultFunctionsPipeline(
+		app.processEdgeXEvent)
+	if err != nil {
+		return errors.Wrap(err, "failed to set default pipeline to processEdgeXEvent")
+	}
+
 	if err := app.queryAllPipelineStatuses(); err != nil {
 		// do not exit, just log
 		app.lc.Errorf("Unable to query EVAM pipeline statuses. Is EVAM running? %s", err.Error())
 	}
 
+	devices, err := app.getAllDevices()
+	if err != nil {
+		app.lc.Errorf("Unable to query all devices")
+
+	} else if devices == nil {
+		app.lc.Debugf("No devices found")
+	} else {
+
+	for _, device := range devices {
+		pipelineRunning := app.isPipelineRunning(device.Name)
+
+		if pipelineRunning {
+			app.lc.Debugf("pipeline is already running for device %s", device.Name)
+			continue
+		}
```

Comment:
you can remove this check as the same code is present in the app.startDefaultPipeline itself.